src/IO/lbsLoadReaderVT.py

```diff
@@ -202,18 +139,56 @@ def read_iteration(self, n_p, time_step):
         # Iterate over all processors
         for p in range(n_p):
             # Read data for given iteration and assign it to processor
-            proc_iter_map = self.read(p, time_step)
+            proc_iter_map = self.read(p, phase_id)
 
             # Try to retrieve processor information at given time-step
             try:
-                procs[p] = proc_iter_map[time_step]
+                procs[p] = proc_iter_map[phase_id]
             except KeyError:
                 print(bcolors.ERR
-                      + "*  ERROR: [LoadReaderVT] Could not retrieve information for processor {} at time_step {}".format(
+                      + "*  ERROR: [LoadReaderVT] Could not retrieve information for processor {} at phase_id {}".format(
                     p,
-                    time_step)
+                    phase_id)
                       + bcolors.END)
                 sys.exit(1)
 
         # Return populated list of processors
         return procs
+
+    def json_reader(self, returned_dict: dict, file_name: str, phase_ids, node_id: int) -> dict:
+        """ Reader compatible with current VT Object Map files (json)
+        """
+        with open(file_name, 'rb') as compr_json_file:
+            compr_bytes = compr_json_file.read()
+            try:
+                decompr_bytes = brotli.decompress(compr_bytes)
+                decompressed_dict = json.loads(decompr_bytes.decode('utf-8'))
+            except brotli.error:
+                decompressed_dict = json.loads(compr_bytes.decode('utf-8'))
+
+        # defining phases from file
+        phases = decompressed_dict['phases']
+        # iterating over phases
+        for phase in phases:
+            phase_id = phase['id']
+            for task in phase['tasks']:
+                task_time = task.get('time')
+                task_object_id = task.get('entity').get('id')
+
+                # Update processor if iteration was requested
+                if phase_ids in (phase_id, -1):
+                    # Instantiate object with retrieved parameters
+                    obj = Object(task_object_id, task_time, node_id)
+
+                    # If this iteration was never encoutered initialize proc object
+                    returned_dict.setdefault(phase_id, Processor(node_id))
+
+                    # Add object to processor
+                    returned_dict[phase_id].add_object(obj)
+
+                    # Print debug information when requested
+                    if self.verbose:
+                        print(f"{bcolors.HEADER}[LoadReaderVT] {bcolors.END}iteration = {phase_id}, "
+                              f"object id = {task_object_id}, time = {task_time}")
+
```

Comment:
We're not reading in any communication edges.  Does ant part of LBAF require them?

Reply:
I haven't provide any additional functionality to LBAF, if it's lacking of some, that is another topic for the meeting.